server/src/main/java/io/druid/segment/realtime/firehose/ReplayableFirehoseFactory.java

```diff
@@ -162,6 +162,11 @@ public ReplayableFirehose(InputRowParser parser) throws IOException
               while (delegateFirehose.hasMore() && cos.getCount() < getMaxTempFileSize()) {
                 try {
                   InputRow row = delegateFirehose.nextRow();
+
+                  if (row == null) {
+                    continue;
```

Comment:
@gianm I realized that this is to avoid marshalling nulls in ReplayableFirehose because jackson seems to not support it. I think we have two options. 
* Removing ReplayableFirehose. It will be remove in #4193 anyway. 
* Skip null rows in ReplayableFirehose as well.

I think the first option is better. What do you think?